### PR TITLE
make: add "none" and "core20" as supported bases.

### PR DESCRIPTION
**Pull request by xnox**
LP: #1833409

Signed-off-by: Dimitri John Ledkov <dimitri.ledkov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/plugins/make.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def get_build_properties(cls):
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
 
-        if project.info.base not in ("core", "core16", "core18"):
+        if project.info.base not in ("none", "core", "core16", "core18", "core20"):
```
Comment (sergiusens) [inline snapcraft/plugins/make.py]: none is still not a base, mind removing it?

Comment (xnox) [reply snapcraft/plugins/make.py]: That depends on what should I use in the core20 snapcraft.yaml.

At the moment, I thought that core20 snap would use "base: none" in snapcraft.yaml and it does use make plugin.

Or should core20 snapcraft declare "base: core20" ?

it sounded like a self-reference to me, but after Gustavo voiced it, it didn't sound that odd.

Comment (sergiusens) [reply snapcraft/plugins/make.py]: #2623 has the right information encoded to make things work.

**Comment by sergiusens** [comment]
Closing this as none is not a base and core20 does not really exist yet.

**Comment by xnox** [comment]
core20 snap is out in edge, and this is now blocking building pc gadget and kernel snaps.

Please merge, and upload to at least edge channel as soon as possible.

**Pull request closed without merging**